### PR TITLE
Restructure infrastructure docs

### LINKED ISSUE #1757: Restructure our infrastructure docs
### Context

We are getting a lot of docs in infrastructure.2i2c.org now, and I think the "how-to" section is becoming a bit of a "catch-all" area, which ultimately makes the docs you need more difficult to find.

### Proposal

I'm proposing we refactor and split up the "how-to" section into the following, separate sections:

* **Hub Deployment Guide**
  * This contains all the documentation required for deploying hubs and enabling/configuring specific features
  * It should follow, more or less, the order of operations required to deploy a hub so an engineer can just step through the docs in order, i.e., start with "Create new cloud accounts" -> "Create a k8s cluster" -> "deploy support infrastructure" -> "deploy a hub" -> and then we get into the realm of turning on/off specific features so this part can be less structured I think
* **Site Reliability Engineering Guide**
  * This contains day-to-day tasks we do _outside_ of hub deployments, e.g., a lot of the content currently under ["Cloud and Kubernetes infrastructure"](https://infrastructure.2i2c.org/en/latest/howto/k8s/index.html) and ["Support"](https://infrastructure.2i2c.org/en/latest/howto/support/index.html)
  * When we have incidents and work out how to resolve them, the solutions should be documented under this guide for the next occurrence (where the solution is not a "this will never happen again" kind of fix, which obviously we should be aiming for)

What does @2i2c-org/tech-team think? I am happy to implement this if we have consensus.

### Updates and actions

New docs:

- [x] https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1285659593
- [x] https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1285890909 (#1806)

Common problems and their solutions (troubleshooting/support):

- [ ] ~~https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1286117676 (https://github.com/2i2c-org/infrastructure/pull/1780#issuecomment-1282096496)~~
- [x] https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1287027429
- [x] https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1292666904 (https://github.com/2i2c-org/docs/issues/100#issuecomment-925962073)

### LINKED ISSUE #1815: Remove reference to pilot-hub image in our docs
@yuvipanda Did this image used to be called `pilot-hub`? I am thinking we may need to

* clean the old repo out of quay.io https://quay.io/organization/2i2c (captured in https://github.com/2i2c-org/infrastructure/issues/1840)

- [x] update the docs here as that still references `pilot-hub`: https://infrastructure.2i2c.org/en/latest/topic/hub-image.html

_Originally posted by @sgibson91 in https://github.com/2i2c-org/infrastructure/pull/1791#discussion_r1004496202_

### PR DESCRIPTION
**Pull request by sgibson91**
fixes #1757 

**New structure:**

- Adds a Site Reliability Engineering guide
- Adds a Hub Deployment Guide
- Splits up the grafana documentation
  - In the Hub Deployment Guide
    - Deploying the support chart itself
    - Deploying the grafana dashboards and setting up CI/CD
    - Registering the cluster-specific prometheus with the central grafana
  - How-to
    - Optionally enable github auth for dedicated-cluster grafanas
  - Topic
    - What is the central grafana
    - How to access grafana
    - How to read the dashboards

**New documentation added:**

- Community-specific CNAMEs (ref: https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1285659593)
- Troubleshooting --> Common problems and their solutions (in SRE guide)
  - How to resize the NFS disk when it is full (ref: https://github.com/2i2c-org/docs/issues/100#issuecomment-925962073)
  - How to reset GitHub OAuth apps when users get a 403 error (ref: https://github.com/2i2c-org/infrastructure/issues/1757#issuecomment-1287027429)

**Updated docs/other fixes:**

- fixes #1815 
- minor styling changes such as consistent casing for headings, adding new labels where appropriate, ensuring tabbed content syncs and uses the most modern dependencies
- Improvements to the new cluster docs
- restructure the topic sections
- group "enable specific hub features" subpages into categories for the howto section
- make it explicit that engineers should get org admin access when setting up GitHub auth
  - ref: https://github.com/2i2c-org/infrastructure/issues/1806

### CODE REVIEW

**Comment by damianavila** [comment]
>  How does the structure look?
> You can view this PR on readthedocs here: https://2i2c-pilot-hubs--1814.org.readthedocs.build/en/1814/

NICE! I really like it.

File: docs/hub-deployment-guide/deploy-support/setup-grafana-dashboards.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Now, we will deploy some specific dashboards to the grafana instance we deployed as part of [](deploy-support-chart).
+
+````{seealso}
+It is also possible to [enable logging in with GitHub](grafana-dashboards:github-auth)
+to allow Community Representatives and Hub Administrators to access these dashboards as well.
+```{warning}
+We should enable GitHub login for _all_ Grafana dashboards running on **dedicated**
+clusters, so the Community Representatives have access to them.
+```
+````
```
Comment (sgibson91) [inline docs/hub-deployment-guide/deploy-support/setup-grafana-dashboards.md:13]: I'm going to go out on a limb here and say that this is a good policy to have regarding grafana access 😄

Comment (yuvipanda) [reply docs/hub-deployment-guide/deploy-support/setup-grafana-dashboards.md:13]: Yes I agree. Long term it might be nice to instead have it use the JupyterHub for auth, but this is easier to do for now :)

**Comment by yuvipanda** [comment]
image

I love this structure, @sgibson91. It makes it easier to *write* documentation as well!

File: docs/hub-deployment-guide/deploy-support/setup-grafana-dashboards.md (hunk 1 above)
Comment (sgibson91) [reply docs/hub-deployment-guide/deploy-support/setup-grafana-dashboards.md:13]: Having just setup this up for the first time, I see a need for pushing forward the Hub OAuth flow. _Unless_ we are fine with either i) _anyone_ being able to "sign up" and access the grafana dashbaords, or ii) also creating a grafana-viewers GitHub team as part of https://github.com/2i2c-org/infrastructure/issues/1806

ref: https://github.com/2i2c-org/infrastructure/pull/1830#discussion_r1007924131

**Comment by GeorgianaElena** [comment]
- >  How does the structure look? 
    You can view this PR on readthedocs here: https://2i2c-pilot-hubs--1814.org.readthedocs.build/en/1814/

   I love the new structure too ❤️ and believe we should adopt it asap.

- Can I propose something to push this PR forward? 🚀 

  I just noticed it changes 61 files, which is lot (it probably touches all of the docs files 2i2c has :D). Can I suggest splitting this PR into smaller ones so that we can merge it more quickly and get to take advantage of the new docs structure? (ref https://github.com/2i2c-org/infrastructure/pull/1817). Right now, it's pretty hard to follow which is new info and what has just been restructured. Some files appear as deleted or as new files, when in reality they've just been moved and then restructured.

  I believe what it would be specifically useful is one PR that changes **only** or mostly the structure. And since there's agreement around that already, I believe that would get merged pretty quickly. 
   And then other PR or PRs for content.
   What do you think?

**Comment by sgibson91** [comment]
I'm very nearly at the end of my checklist for this PR and expect to mark it ready for review very soon. I also think extracting already-pushed commits from this branch will get me in a git mess that will likely slow the process down instead of speeding it up. So I'm not confident.

**Review by yuvipanda — APPROVED** [review]
Two minor comments, but otherwise let's get this merged!

THANK YOU SO MUCH @sgibson91 for doing this kinda structurally deeply important work!

File: docs/howto/manage-domains/set-cnames.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+You can use the template below to comment on the relevant "New Hub - Request
+deployment" issue asking them to complete the steps.
+
+```markdown
+### Instructions for setting CNAMEs
+
+For the hub to be available at hub.`<community-domain>`, please follow the below steps:
+
+1. Please create a CNAME record in your DNS zone
+2. Call the CNAME record `hub`
```
[the remark this answers is not in the document]
Comment (sgibson91) [reply docs/howto/manage-domains/set-cnames.md]: Addressed in [`12d118e` (#1814)](https://github.com/2i2c-org/infrastructure/pull/1814/commits/12d118e6f019238e610b003f375a4a776f7bcaba)

**Comment by damianavila** [comment]
Indeed, wonderful work @sgibson91!!!
